pkg/webhook/admission/defaulter_handler.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	goerrors "errors"
+	"net/http"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// WithDefaulter defines functions for setting defaults on resources.
+type WithDefaulter interface {
+	Default(ctx context.Context, obj runtime.Object) error
+}
+
+// WithCustomDefaulter creates a new Webhook for a WithDefaulter interface.
+func WithCustomDefaulter(obj runtime.Object, defaulter WithDefaulter) *Webhook {
+	return &Webhook{
+		Handler: &defaulterForType{handler: defaulter},
+	}
+}
+
+type defaulterForType struct {
+	handler WithDefaulter
```

Comment:
```suggestion
	defaulter WithDefaulter
```
nit, Not sure if it's worth it but we could also call this defaulter to keep it more similar to the other defaulter implementation (but it's probably also the better name)